kinto/views/permissions.py

```diff
@@ -82,6 +82,17 @@ def get_records(self, filters=None, sorting=None, pagination_rules=None,
         # Check settings for every allowed resources.
         from_settings = allowed_from_settings(self.request.registry.settings, principals)
 
+        # Add additional resources and permissions defined in settings/plugins
+        bucket_perms = from_settings.get('')
+        if bucket_perms:
+            for bucket_perm in bucket_perms:
+                resource_name, _ = bucket_perm.split(':')
+                uri = '/{0}s'.format(resource_name)
```

Comment:
This will work, but if we want to be completely safe here, it should be:
```py
uri = core_utils.strip_uri_prefix(self.request.route_path(f'{resource_name}-collection'))
```